### PR TITLE
Feature/kak/indicator params cleanup

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Assorted small tidying made while investigating #182.

 - make `IndicatorParameter` a modeled type
 - move the query parameters (as opposed to path parameters) to a nested object

## Testing

 - shouldn't break anything :woman_shrugging:

### CODE REVIEW

File: src/app/models/indicator-query-opts.model.ts [hunk 1]
```diff
@@ -7,8 +7,10 @@ export interface IndicatorQueryOpts {
     indicator: Indicator;
     city: City;
     scenario: Scenario;
-    climateModels?: ClimateModel[];
-    years?: string[];
-    time_aggregation?: string;
-    unit?: string;
+    params: {
```
Comment (fungjj92) [inline src/app/models/indicator-query-opts.model.ts]: Hm these nested in params diverges from the API response: 

screen shot 2017-08-09 at 5 10 29 pm

Maybe you were looking at indicator.parameters ?

**Comment by fungjj92** [comment]
Adding the model for IndicatorParam is good though as we'll expect to use that data

File: src/app/models/indicator-query-opts.model.ts (hunk 1 above)
Comment (flibbertigibbet) [inline src/app/models/indicator-query-opts.model.ts]: There are the query parameters passed to make the request and not intended to map to the response. Intention here is to distinguish them from the *path* parameters, which are the other things on this object.

Comment (fungjj92) [inline src/app/models/indicator-query-opts.model.ts]: Oh I see now. Sounds good!

File: src/app/models/indicator-parameter.model.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/* tslint:disable:variable-name */
```
Comment (CloudNiner) [inline src/app/models/indicator-parameter.model.ts]: Do any of these params actually fail the variable name check? It's probably better to only disable checks when necessary, right?

Comment (flibbertigibbet) [inline src/app/models/indicator-parameter.model.ts]: No actually, was just defined on several other models.

**Pull request merged**